server/src/main/scala/org/http4s/server/middleware/ResponseLogger.scala

```diff
@@ -67,7 +67,7 @@ object ResponseLogger {
         }
         .guaranteeCase {
           case ExitCase.Error(t) => fk(log(s"service raised an error: ${t.getClass}"))
-          case ExitCase.Canceled => fk(log(s"service cancelled response for request [$req]"))
+          case ExitCase.Canceled => fk(log(s"service canceled response for request"))
```

Comment:
Suggestion, per https://github.com/http4s/http4s/issues/3482#issuecomment-641998796, log `"service canceled response for request: ${req.method} ${req.pathInfo}."` rather than throw away all details of the request via `"service canceled response for request"`.

Regarding:

> But we'd also be fighting against people who put a request ID in their headers and are using that to correlate the logs.

Or, if you modify the `toString` of `Request`, perhaps you could include headers' values that are definitely safe to log? Does such a list of universally non-sensitive HTTP Request Headers exist?

Reply:
The problem is that it's not a `Request`, it's an `A`.  We can't refer to method and pathInfo here without pattern matching.  If we pattern match, we need to handle `ContextRequest`.  If we handle `ContextRequest`, it raises questions about other request wrappers.  The only failsafe tools we have are to:

* Not log anything about `A`
* Log `A`'s type. Not useful.
* Log `A`'s .toString, and change `Request`'s `.toString` definition. Nobody had an appetite for this.

The default `Request#toString` does redact a few well-known headers, but many people, including the issue reporter, have additional sensitive headers.  We can't get them all without some global system property, which, no.

Reply:
Thanks, @rossabaker. Given the risk of https://github.com/http4s/http4s/issues/3482, is it better to use this PR's updated `String` instead of possibly leaking sensitive headers' values?

In other words, it seems to me that merging this PR is worth it to avoid logging sensitive headers' values.

Thoughts?

Reply:
I am fine with either this approach or changing `Request#toString` and picked the former just to get a fix out there.

I'm wary of pattern matching, which only makes the leak more rare without actually fixing it.